src/snowflake/cli/_plugins/nativeapp/v2_conversions/v2_to_v1_decorator.py

```diff
@@ -48,28 +51,37 @@ def _convert_v2_artifact_to_v1_dict(
     return v2_artifact
 
 
-def _pdf_v2_to_v1(v2_definition: DefinitionV20) -> DefinitionV11:
+def _pdf_v2_to_v1(
+    v2_definition: DefinitionV20,
+    package_entity_id: str = "",
+    app_entity_id: str = "",
```

Comment:
Should we enforce the expected type of these referenced entities?

Reply:
you mean fail if we find an entity with that ID but of a different type?

Reply:
this is done in a slightly different way; the new logic checks that the ID passed refers to the correct package  or app and fails if it doesn't find one, which would also happen if someone passed in an ID for an entity of a different type